experimental/traffic-portal/src/app/api/cdn.service.ts

```diff
@@ -109,4 +109,16 @@ export class CDNService extends APIService {
 
 		return this.put<ResponseCDN>(`cdns/${id}`, body).toPromise();
 	}
+
+	/**
+	 * Queues or dequeues updates on a CDN's servers.
+	 *
+	 * @param cdn The CDN to queue or dequeue updates on.
+	 * @param action The action to perform on the CDN, either "queue" or "dequeue".
+	 */
+	public async queueCDNUpdates(cdn: ResponseCDN, action: "queue" | "dequeue"): Promise<CDNQueueResponse> {
```

Comment:
Ideally you should be able to accept an ID anywhere a response structure is accepted and vice-versa

Reply:
I know what you mean, but accepting both and checking for either both ID and `ResponseCDN` iinstance in all places is not the way to do it, IMO, because it is extra work, and the point of Typescript is to eliminate ambiguities like that one.

Maybe a future improvement having it take a new `CDN` type that *can* have as much info as `ResponseCDN` but only requires an `ID`, and it's only resolved by the API (or some other way to defer its resolution)?

If something like that is acceptable, then IMO we should do that in the places we're currently accepting either an object or ID.

Reply:
Fair enough, made it support either ID or `ResponseCDN` in e462698155